### DISCUSSION TITLE
Why is it distributed as a compiler plugin?

### DISCUSSION THREAD

**Discussion by joakimtall** [cashapp/molecule#156] [Q&A] [answered]
I looked at the source and it seems it pretty much only adds a normal dependency so I got curious, why distribute it as a compiler plugin? Genuinely curious, not complaining 👍

**Comment by JakeWharton** [comment, marked as answer]
It adds two dependencies: the runtime APIs as a normal dependency and the Jetpack Compose Kotlin compiler plugin. It also enables the stub generation behavior of the Compose compiler plugin when targeting Kotlin/JS.

It's the same reason that JetBrains Compose and Jetpack Compose (via AGP) are Gradle plugins. There's multiple things to set up.
